sdv/constraints/tabular.py

```diff
@@ -475,3 +476,146 @@ def reverse_transform(self, table_data):
+    is a one hot vector. During the reverse transform, the output of the model
+    is transformed so that the column with the largest value is set to 1 while
+    all other columns are set to 0.
+
+    Args:
+        columns (list[str]):
+            Names of the columns containing one hot rows.
+        handling_strategy (str):
+            How this Constraint should be handled, which can be ``transform``
+            or ``reject_sampling`` (not recommended). Defaults to ``transform``.
+    """
+
+    def __init__(self, columns, handling_strategy='transform'):
+        self._columns = columns
+        self.constraint_columns = tuple(columns)
+        super().__init__(handling_strategy, fit_columns_model=True)
+
+    def _sample_constraint_columns(self, table_data):
+        """Handle constraint columns when conditioning.
+
+        When handling a set of one-hot columns, a subset of columns may be provided
+        to condition on. To handle this, this function does the following:
+
+        1. If the user specifies that a particular column must be 1,
+           then all other columns must be 0.
+
+        2. If the user specifies that one or more columns must be 0, then
+           we need to sample the other columns and select the highest value
+           and enforce the one-hot constraint.
+
+        3. If the user specifies something invalid, we need to raise an error.
+
+        Args:
+            table_data (pandas.DataFrame):
+                Table data containing the conditions.
+
+        Returns:
+            pandas.DataFrame:
+                Table data with the constraint columns filled in.
+
+        Raise:
+            `ValueError` if the conditions are invalid.
+        """
+        table_data = table_data.copy()
+
+        condition_columns = [col for col in self._columns if col in table_data.columns]
+        if not table_data[condition_columns].isin([0.0, 1.0]).all(axis=1).all():
+            raise ValueError('Condition values must be ones or zeros.')
+
+        if (table_data[condition_columns].sum(axis=1) > 1.0).any():
+            raise ValueError('Each row of a condition can only contain one number one.')
+
+        has_one = table_data[condition_columns].sum(axis=1) == 1.0
+        if (~has_one).sum() > 0:
+            sub_table_data = table_data.loc[~has_one, condition_columns]
+
+            if len(condition_columns) == len(self._columns) - 1:
+                proposed_table_data = sub_table_data.copy()
+                for column in self._columns:
+                    if column not in condition_columns:
+                        proposed_table_data[column] = 1.0
+
+            else:
+                condition_columns = [
+                    c for c in self.constraint_columns if c in sub_table_data.columns
+                ]
+                grouped_conditions = sub_table_data[condition_columns].groupby(condition_columns)
+                all_sampled_rows = list()
+                for group, df in grouped_conditions:
```

Comment:
It's actually not necessary, because the grouped conditions are always 0.0. I changed the logic to remove the grouping in the latest version. 

**Note:** the constraint base class doesn't seem to implement any logic for keeping track of the order, I think you are referring to the tabular base class. After some exploring, I think we actually need to add the logic to the constraint base class, I don't see any test cases which test this specifically so it's possible it's currently broken...

Reply:
> It's actually not necessary, because the grouped conditions are always 0.0. I changed the logic to remove the grouping in the latest version.

Oh, of course! :-)

> **Note:** the constraint base class doesn't seem to implement any logic for keeping track of the order, I think you are referring to the tabular base class. After some exploring, I think we actually need to add the logic to the constraint base class, I don't see any test cases which test this specifically so it's possible it's currently broken...

Very good point. Maybe we can open an issue to just "review" this? If we look at it and it turns out it is not a problem we will end up closing it without doing anything.